cpp/oneapi/dal/algo/pca/infer_types.cpp

```diff
@@ -0,0 +1,83 @@
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/algo/pca/infer_types.hpp"
+#include "oneapi/dal/detail/common.hpp"
+#include "oneapi/dal/exceptions.hpp"
+
+namespace oneapi::dal::pca {
+
+template <typename Task>
+class detail::infer_input_impl : public base {
+public:
+    infer_input_impl(const model<Task>& trained_model, const table& data)
+            : trained_model(trained_model),
+              data(data) {}
+    model<Task> trained_model;
+    table data;
+};
+
+template <typename Task>
+class detail::infer_result_impl : public base {
+public:
+    table transformed_data;
+    double objective_function_value = 0.0;
```

Comment:
1) What does `objective_function` mean for PCA?
2) Why it has `double` type?

Reply:
fixed